vividus-extension-selenium/src/main/java/org/vividus/ui/action/search/ShadowDomSearchParameters.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2019-2021 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.vividus.ui.action.search;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+
+public class ShadowDomSearchParameters extends SearchParameters
+{
+    private List<String> shadowHosts;
+
+    public ShadowDomSearchParameters(List<String> shadowHosts, String value)
+    {
+        super(value);
+        this.shadowHosts = shadowHosts;
+    }
+
+    public ShadowDomSearchParameters(List<String> shadowHosts, String value, Visibility visibility)
+    {
+        super(value, visibility);
+        this.shadowHosts = shadowHosts;
+    }
+
+    public ShadowDomSearchParameters(List<String> shadowHosts, String value, Visibility visibility,
+            boolean waitForElement)
+    {
+        super(value, visibility, waitForElement);
+        this.shadowHosts = shadowHosts;
+    }
+
+    public String getUpperShadowHost()
+    {
+        return shadowHosts.get(0);
+    }
+
+    public String[] getInnerShadowHosts()
+    {
+        return Arrays.copyOf(shadowHosts.stream().skip(1).toArray(), shadowHosts.toArray().length - 1, String[].class);
```

Comment:
what about moving this logic to constructor? so that we have `upperShadowHost` and `innerShadowHosts` fileds